src/controlplane/components.go

```diff
@@ -26,7 +23,9 @@ type Component struct {
 	UseMTLSAuthentication           bool
 	Specs                           definition.SpecGroups
 	Queries                         []prometheus.Query
-	Labels                          []labels
+	Labels                          []componentLabels
+	// TODO this will be refactored with config per label group.
+	Namespace string
```

Comment:
In this file WithAPIServerSecurePort is not unused, is it intended? If so we should delete it

Reply:
yes all this will be removed on the next refactor when using the new config schema, I already removed all this func in the next branch :)

Reply:
just removed from here too